addons/mail/static/src/messaging/entity/entity/field.js

```diff
@@ -0,0 +1,653 @@
+odoo.define('mail.messaging.entity.Field', function (require) {
+'use strict';
+
+/**
+ * Class whose instances represent field parts of an entity instance. Processing
+ * of get/set of a field of an entity instance is handled here.
+ */
+class Field {
+
+    /**
+     * @param {mail.messaging.entity.Entity} entity
```

Comment:
It would probably be better if there was one field instance per field and per entity class (as opposed to per entity instance). Then all methods here need to be adapted to take as parameter the "current entity instance" on which to apply the operation.